### ISSUE TITLE
Reactor only firing once

### ISSUE THREAD

**Issue by dvogt** [saltstack/salt#11693]
I have a problem with a reactor firing of a runner only once. 
If I restart the salt master, I can get the reactor to work once. 
If I try a second time without restarting, the reactor never fires.
However a message is received on the message bus. 

On the minion, I fire a message:
```
sudo salt-call event.fire_master 'Deploy' 'deploy/qa'
```

On the master I receive a message:
```
*************************
Tag: deploy/qa
Data:
{'_stamp': '2014-04-01_05:31:59.939744',
 'cmd': '_minion_event',
 'data': 'Deploy',
 'id': 'build',
 'tag': 'deploy/qa'}
Event fired at Tue Apr  1 05:32:01 2014 
*************************
```

After the first run, I get this error:
```
Exception AttributeError: "'ReqServer' object has no attribute 'work_procs'" in <bound method ReqServer.__del__ of <salt.master.ReqServer object at 0x16a7310>> ignored
Exception TypeError: 'an integer is required' in <bound method SaltEvent.__del__ of <salt.utils.event.SaltEvent object at 0x1008d10>> ignored
```

------------------------

On the master I can run salt-run several time without any issues. 
```
salt-run mydeploy.deploy qa
```

Another observation is that even though I can see the event fired on the eventlistener, there is no attempt by the master to take action. 
There is no output while running "salt-master -l debug"

When running: salt-run mydeploy.deploy qa
I see that the runner is doing what it supposed to. 

------------------------

The reactor conf, the state file and the sample code can be found at:
http://pastebin.com/Ecfhx425

The runner is actually more complicated than the example. 

Environment:
```
Ubuntu 12.04
salt --versions-report
           Salt: 2014.1.0
         Python: 2.7.3 (default, Feb 27 2014, 19:58:35)
         Jinja2: 2.6
       M2Crypto: 0.21.1
 msgpack-python: 0.1.10
   msgpack-pure: Not Installed
       pycrypto: 2.4.1
         PyYAML: 3.10
          PyZMQ: 13.0.0
            ZMQ: 3.2.2
```

**Comment by cachedout** [comment]
Yes, calling exit inside a runner would have caused the master process to die. I wrote a fix for this somewhere. I will see if I can find it.

**Comment by cachedout** [comment]
Fixed by #19558
